### PR TITLE
[Ready to MERGE] Swagger for Tenant Properties API

### PR DESCRIPTION
**Pull request by santoshsinha100**
added swagger for Tenant Properties

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json
Comment (nschonni) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: Trim?

Comment (santoshsinha100) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: done

Comment (nschonni) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: :+1:  looks like you replaced the next line indentation with a tab though :wink:

**Comment by santoshsinha100** [comment]
I am getting below error while trying to run the checks on GitHub after billing.json update. However all the remaining checks and OAV  test were performed and passed. Can you please help me on this.

Here’s the ARM Manifest update PR. I have created for “DiscoverTenant” api.
https://armmanifest.visualstudio.com/One/_git/Manifest/pullrequest/468?_a=files

> AutoRest code generation utility [version: 2.0.4283; node: v8.12.0]
> (C) 2018 Microsoft Corporation.
> https://aka.ms/autorest
>    Loading AutoRest core      '/root/.autorest/@microsoft.azure_autorest-core@2.0.4289/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4289)
>    Loading AutoRest extension '@microsoft.azure/autorest.java' (2.1.71->2.1.71)
>    Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.38->2.3.38)
> Processing batch task - {"tag":"package-2017-04-preview"} .
> Processing batch task - {"tag":"package-2018-03-preview"} .
> FATAL: System.Exception: Duplicate File Generation: src/main/java/com/microsoft/azure/management/billing/v2018_03_01_preview/TenantProperties.java
>    at AutoRest.Core.CodeGenerator.<Write>d__13.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at AutoRest.Core.CodeGenerator.<Write>d__12.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at AutoRest.Java.Azure.Fluent.CodeGeneratorJvaf.<Generate>d__6.MoveNext() in /opt/vsts/work/1/s/src/azurefluent/CodeGeneratorJvaf.cs:line 106
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at AutoRest.Java.Program.<ProcessInternal>d__3.MoveNext() in /opt/vsts/work/1/s/src/Program.cs:line 114
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at NewPlugin.<Process>d__15.MoveNext()
> FATAL: java/generate - FAILED
> FATAL: Error: Plugin java reported failure.
> Process() cancelled due to exception : Plugin java reported failure.
> Failure during batch task - {"tag":"package-2018-03-preview"} -- Error: Plugin java reported failure..
>   Error: Plugin java reported failure.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json
Comment (santoshsinha100) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: I don't see any tab in notepadd ++.  Can you please approve it else it won't get published as ARM api.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json
Comment (santoshsinha100) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: build issue is fixed now. Waiting for approval.

Comment (santoshsinha100) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: Regarding Name "DiscoverTenants" its actually doing a tenant discovery in Power BI custom connector and this API is designed only for that purpose. I hope this justifies your query.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json
Comment (KrisBash) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: As billingProfileId is mandatory, this is likely better served by a POST action request where the billingProfileId is passed in the request body. See this for an [example ](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#subscription-wide-reads-and-actions)

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comment

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json
Comment (santoshsinha100) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: This API is designed for Power BI custom connector to achieve multi-tenancy.  This design is approved by connector Team.  It is a request flow which brings Tenant Id for a specified billingProfileId input. API is doing a tenant discovery and that's why it is named as "DiscoverTenant". I hope this clarifies your question.

Comment (santoshsinha100) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: Changes in swagger are committed to making the URL compliant with the Resource Provider Contract. The new URL is
'/providers/Microsoft.Billing/billingProfiles/{billingProfileId}/discoverTenants'

I am keeping the name discoverTenants as the sole purpose of this API is to discover the Tenants only. As all the review comments are implemented can you please mark this approve.

Comment (KrisBash) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: thank you for these changes. This path is syntactically correct. The resource type name in this case would be "discoverTenants" which is a bit odd, as verbs are atypical in a resource type name. I would prefer it to be named with a noun pattern for consistency. `discoveredTenants` would be substantially preferable in my opinion.

**Review by KrisBash — APPROVED** [review]
Signing off with comment. Please do review the comment though.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json
Comment (santoshsinha100) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/billing.json]: we are moving our new API "discoverTenants" from "Microsoft.Billing" to "Microsoft.Consumption". All the change we made previously remains the same except for different folder and path.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json
Comment (santoshsinha100) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json]: We are moving our new API "discoverTenants" from "Microsoft. Billing" to "Microsoft.Consumption". All the change we made previously remains the same except for different folder and path.  There is no change in the URL pattern we agreed in our previous discussion.

Comment (santoshsinha100) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json]: There is change in namespace and addition of one extra parameter "billingAccountId".

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json
Comment (santoshsinha100) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json]: Change in Namespace followed by one additional parameter "billingAccountId"

**Review by praries880 — APPROVED** [review]

**Comment by praries880** [comment]
@santoshsinha100  do let us lknow when its safe to merge these  changes in

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json
Comment (santoshsinha100) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json]: Changes in tenant response as per ARM recommendations.

Comment (santoshsinha100) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/consumption.json]: Also changes in route from "discoverTenants" to "tenants" now.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/examples/TenantsGet.json
Comment (RyanBensonMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/examples/TenantsGet.json]: the body payload should match the definition of TenantListResult

Comment (santoshsinha100) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/examples/TenantsGet.json]: its done now.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/examples/TenantsGet.json
Comment (RyanBensonMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-10-01/examples/TenantsGet.json]: Your id currently references the collection itself, we want to mimic a collection api who has unique resources in it. Your id should end with the 'name' of the resource, in your case 'tenant1'
"id": "...Microsoft.Consumption/tenants/tenant1",

**Review by ryansbenson — COMMENTED** [review]
Discussed offline and Santosh will be moving the /tenants api to a new api version before approval

**Review by ryansbenson — COMMENTED** [review]
We will be leaving the tenants API in 2018-10-01 as defined in this PR

**Review by praries880 — APPROVED** [review]

**Comment by santoshsinha100** [comment]
This item is long due and ready to merge. Can you please merge this item to close an open item in our 360 dashboards. All the comments are resolved. Arm manifest was reviewed and got successfully updated. 

Please close this open item.

https://msazure.visualstudio.com/Unified%20Platform%20KPIs/_workitems/edit/3598659

**Comment by praries880** [comment]
@santoshsinha100 You don't seem to have signed the CLA... can you kindly click on the `Details` link for the one pending CI check and sign it?

**Pull request merged**
